kolibri/plugins/slideshow_render/assets/src/views/SlideshowRendererComponent.vue

```diff
@@ -0,0 +1,214 @@
+<template>
+
+  <CoreFullscreen
+    ref="slideshowRenderer"
+    class="slideshow-renderer"
+    @changeFullscreen="isInFullscreen = $event"
+  >
+    <UiIconButton
+      class="btn"
+      :ariaLabel="isInFullscreen ? $tr('exitFullscreen') : $tr('enterFullscreen')"
+      color="primary"
+      size="small"
+      @click="$refs.slideshowRenderer.toggleFullscreen()"
+    >
+      <mat-svg v-if="isInFullscreen" name="fullscreen_exit" category="navigation" />
+      <mat-svg v-else name="fullscreen" category="navigation" />
+    </UiIconButton>
+    <Hooper v-if="slides" @slide="handleSlide">
+      <Slide v-for="slide in slides" :key="slide.id">
+        <div
+          class="slideshow-slide-image"
+          :style="{
+            height: `calc(100% - ${caption_height}px)`
+          }"
+        >
+          <img
+            :src="slide.storage_url"
+            :aria-labelledby="slideTextId(slide.id)"
+          >
+        </div>
+        <div :id="slideTextId(slide.id)" class="hidden-descriptive-text">
+          {{ slide.descriptive_text || slide.caption }}
+        </div>
+        <div :ref="slide.id" class="caption">
+          {{ slide.caption }}
+        </div>
+      </Slide>
+      <HooperNavigation
+        slot="hooper-addons"
+        :class="{'hooper-navigation-fullscreen' : isInFullscreen}"
+      />
+      <HooperPagination slot="hooper-addons" />
+    </Hooper>
+  </CoreFullscreen>
+
+</template>
+
+
+<script>
+
+  import _ from 'lodash';
+
+  import themeMixin from 'kolibri.coreVue.mixins.themeMixin';
+  import contentRendererMixin from 'kolibri.coreVue.mixins.contentRendererMixin';
+  import responsiveElement from 'kolibri.coreVue.mixins.responsiveElement';
+  import responsiveWindow from 'kolibri.coreVue.mixins.responsiveWindow';
+
+  import UiIconButton from 'kolibri.coreVue.components.UiIconButton';
+  import CoreFullscreen from 'kolibri.coreVue.components.CoreFullscreen';
+  import {
+    Hooper,
+    Slide,
+    Navigation as HooperNavigation,
+    Pagination as HooperPagination,
+  } from 'hooper';
+  import '../../styles/custom-hooper.css';
+
+  export default {
+    name: 'SlideshowRendererComponent',
+    components: {
+      UiIconButton,
+      CoreFullscreen,
+      Hooper,
+      Slide,
+      HooperPagination,
+      HooperNavigation,
+    },
+    mixins: [contentRendererMixin, themeMixin, responsiveElement, responsiveWindow],
+    props: {
+      defaultFile: {
+        type: Object,
+        default: () => {},
+      },
+      files: {
+        type: Array,
+        default: () => [],
```

Comment:
whenever props are complex types (objects or arrays) we should also provide a `validator` function

Reply:
Thanks for letting me know - I've added validators to both props. Do you think they cover what I'll need to validate well enough?

Reply:
I guess I'm actually still left a bit confused as to what form of data these props expect. From the validators I see a schema like:

```
defaultFile: { storage_url: "*.json" }
```

and 

```
files: [
  { storage_url: "*" },
  //...
]
```

but then down below I also see a reference to `file.preset`.

I guess what I would expect to see in either comments, validator function, or both is a complete description of the shape of the data for the things that the component is calling a "file" (or maybe more accurately a "file descriptor object"?).